pmd-apex/src/test/resources/net/sourceforge/pmd/lang/apex/rule/security/xml/ApexSharingViolations.xml

```diff
@@ -99,6 +99,63 @@ public inherited sharing class MyClass {
    public List<Contact> getAllTheSecrets() {
        return [SELECT Name FROM Contact];
    }
+}
+     ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>Apex test classes do not need a sharing declaration</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+@isTest public class MyClass {
+   public List<Contact> getAllTheSecrets() {
+       return [SELECT Name FROM Contact];
+   }
+}
+     ]]></code>
+    </test-code>
+
+    <!-- @isTest can only be declared on the outer class, it is inherited by the inner class-->
+    <test-code>
+        <description>Apex inner test classes do not need a sharing declaration</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+@isTest public class Outer {
+   public List<Contact> getAllTheOuterSecrets() {
+       return [SELECT Name FROM Contact];
+   }
+   public class Inner {
+       public List<Contact> getAllTheInnerSecrets() {
+           return [SELECT Name FROM Contact];
+       }
+   }
+}
+     ]]></code>
+    </test-code>
+
+    <test-code>
```

Comment:
Found while regression testing against public repos. This was found here https://github.com/SalesforceLabs/Schedul-o-matic-9000/blob/master/force-app/main/default/classes/Dao.cls#L14